benchmark/inference.py

```diff
@@ -0,0 +1,149 @@
+import numpy as np
+import concurrent.futures
+from typing import Tuple
+import time
+from tqdm import tqdm
+from concurrent.futures import wait
+import time
+
+import numpy as np
+import pandas as pd
+from sklearn.model_selection import train_test_split
+from tensorflow.keras.models import load_model
+
+import tensorflow as tf
+import autokeras as ak
+
+import wandb
+from wandb.keras import WandbCallback
+
+def train_model(x_train: np.ndarray, y_train: np.ndarray) -> ak.TextRegressor:
+
+    wandb.init(project="ml-in-prod")
+
+    wandb.config = {
+       "learning_rate": 0.001,
+       "epochs": 5,
+       "batch_size": 128
+    }
+
+    input_node = ak.TextInput()
+    output_node = ak.TextToIntSequence()(input_node)
+    output_node = ak.Embedding()(output_node)
+    # Use separable Conv layers in Keras.
+    output_node = ak.ConvBlock(separable=True)(output_node)
+    output_node = ak.RegressionHead()(output_node)
+    reg = ak.AutoModel(
+        inputs=input_node, outputs=output_node, overwrite=True, max_trials=1, max_model_size=1000000000,
+    )
+
+    reg.fit(x_train, y_train, epochs=5, callbacks=[WandbCallback()])
+
+    model = reg.export_model()
+
+    try:
+        model.save("model_autokeras", save_format="tf")
+    except:
+        model.save("model_autokeras.h5") 
+    
+    art = wandb.Artifact("model_autokeras", type="model")
+    art.add_dir("model_autokeras")
+    wandb.log_artifact(art)
+
+    return reg
+
+
+def get_data(path: str = './test.csv') -> Tuple[np.ndarray, np.ndarray, np.ndarray]:
+    train_dataset = pd.read_csv(path,
+                                na_values="?", comment='\t',
+                                sep=",", skipinitialspace=True)
+
+    X = train_dataset.excerpt
+    y = train_dataset.target
+
+    train_X, val_X, train_y, val_y = train_test_split(X, y, random_state=1)
+
+    train = pd.DataFrame(train_X)
+    val = pd.DataFrame(train_y)
+    test = pd.DataFrame(val_X)
+
+    train.to_csv("train.csv")
+    val.to_csv("val.csv")
+    test.to_csv("test_val.csv")
+
+    return np.array(train_X), np.array(train_y), np.array(val_X)
+
+
+def predict(model: ak.TextRegressor, x: np.ndarray) -> np.ndarray:
+    # replace with real model
+    # time.sleep(0.001)
+    return model.predict(x)
+
+
+def run_inference(x_test: np.ndarray, path:str = "model_autokeras", batch_size: int = 64) -> np.ndarray:
+    wandb.init(project="ml-in-prod")
+
+    wandb.config = {
+       "learning_rate": 0.001,
+       "epochs": 5,
+       "batch_size": 128
+    }
+
+    artifact = wandb.use_artifact("securims/ml-in-prod/model_autokeras:latest", type="model")
+    artifact_dir = artifact.download(root=path)
+
+    model = load_model(path, custom_objects=ak.CUSTOM_OBJECTS)
+
+    y_pred = []
+    for k in range(0, 100):
```

Comment:
why do we need this for loop here?

Reply:
This is just to have enough load on the CPU so that parallelization has at least some benefits